proposals/readonly-instance-members.md

```diff
@@ -143,6 +143,23 @@ public readonly int Prop7 { get; set; }
 public int Prop8 { get; readonly set; }
 ```
 
+Readonly can be applied to manually-implemented events, but not field-like events. Readonly cannot be applied to individual event accessors (add/remove).
+```cs
+// Allowed
+public readonly event Action<EventArgs> Event1
+{
+    add { }
+    remove { }
+}
+
+// Not allowed
+public readonly event Action<EventArgs> Event2;
+public event Action<EventArgs> Event3
```

Comment:
Why would this latter one not be allowed?

Say you have a partially mutable struct, where you support adding more event subscriptions, but not removing them. In which case `add` mutates, but `remove` does not.

Reply:
As an example:
```csharp
public struct B
{
    private Action<EventArgs> _event;
    
    public event Action<EventArgs> Event1
    {
        add { _event += value; }
        readonly remove { }
    }
}
```

Reply:
Mainly because it's easier to start by disallowing it and a less common case than property accessors. I'd suggest revisiting if we get feedback from people who would really benefit from being able to specify different `readonly`-ness on the add and remove accessors.

Reply:
Fair enough.

I'm guessing supporting this is "more difficult" for event accessors, since they don't currently (afair) allow attached modifiers?

Reply:
Yeah, event accessors don't allow any modifiers right now, and it's required to declare both `add` and `remove` accessors. It was feeling like the value added needed to be more clear in order to get past that friction.